### ISSUE TITLE
Allow disabling components of test-only apps on ADB mode

### ISSUE THREAD

**Issue by MuntashirAkon** [MuntashirAkon/AppManager#511]
Is it possible to check a package is testOnly then add a option to disable its component? I think it is useful for "adb user" when they have to use some app with tracker because sometime make app testOnly and reinstall it is possible.

_Originally posted by @d4rkk3y in https://github.com/MuntashirAkon/AppManager/discussions/510#discussioncomment-985418_

**Comment by MuntashirAkon** [comment]
Thanks for your interest!

App Manager currently offers a feature called “block components” for root users only because components can be disabled/blocked only with root. However, if the app is a test-only app, components of the app can be disabled with ADB. The task here is to allow disabling components for the test-only apps if the user has enabled ADB mode of operation.

That being said, your tasks are as follows:
1. In the components tabs of the app details page, if the app is set to test-only and ADB is enabled, reveal the block/unblock button.
2. In the 1-click ops page, enable “block/unblock trackers” and “block components...” for ADB users too, and filter the list with test-only the apps (for ADB users only, others should be the same as it is).
3. In the ComponentsBlocker class, disable applying Intent Firewall (IFW) rules for ADB users.

You can check if ADB is enabled anywhere in the app via `AppPref#isAdbEnabled()`. You do not have to worry about enabling/disabling components as they are already present in the ComponentsBlocker class.

While the tasks are very easy, you might have to analyse the code and read the docs in order to understand the tasks a bit better. You can always ask me here if you have any queries. Good luck!

**Comment by SNNafi** [comment]
Thanks for your quick reply. But another error occurred. :( Didn't find anything useful on the internet. 

```
No matching variant of dev.rikka.tools.refine:gradle-plugin:3.0.3 was found. The consumer was configured to find a runtime of a library compatible with Java 8, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '7.1.1' but:
```

**Comment by SNNafi** [comment]
Sorry for the late update. I have checked task no 2., and it is working fine. 

For task 3, I think changing this code,
```
 private void retrieveDisabledComponents() {
        if (!AppPref.isRootEnabled()) return;
                           --------
}
```
to

```
 private void retrieveDisabledComponents() {
        if (!AppPref.isRootOrAdbEnabled()) return;
                             ----------------------
}
```
do the work? Is it ok?

**Comment by MuntashirAkon** [comment]
Implemented in e9da575b

**Comment by sobaee** [comment]
Since this mod has implemented in the latest version

Where can I found the test-only version of app manager which I can install it by: adb install -t package-name?
